catalog/resource_sql_table.go

```diff
@@ -0,0 +1,383 @@
+	StorageCredentialName string            `json:"storage_credential_name,omitempty" tf:"force_new"`
+	ViewDefinition        string            `json:"view_definition,omitempty"`
+	Comment               string            `json:"comment,omitempty"`
+	Properties            map[string]string `json:"properties,omitempty" tf:"computed"`
+	ClusterID             string            `json:"cluster_id,omitempty" tf:"computed"`
+
+	exec common.CommandExecutor
+}
+
+type SqlTablesAPI struct {
+	client  *common.DatabricksClient
+	context context.Context
+}
+
+func NewSqlTablesAPI(ctx context.Context, m any) SqlTablesAPI {
+	return SqlTablesAPI{m.(*common.DatabricksClient), context.WithValue(ctx, common.Api, common.API_2_1)}
+}
+
+func (a SqlTablesAPI) getTable(name string) (ti SqlTableInfo, err error) {
+	err = a.client.Get(a.context, "/unity-catalog/tables/"+name, nil, &ti)
+	return
+}
+
+func (ti *SqlTableInfo) FullName() string {
+	return fmt.Sprintf("%s.%s.%s", ti.CatalogName, ti.SchemaName, ti.Name)
+}
+
+type SqlTables struct {
+	Tables []SqlTableInfo `json:"tables"`
+}
+
+// These properties are added automatically
+// If we do not customize the diff using these then terraform will constantly try to remove them
+// `properties` is essentially a "partially" computed field
+// This needs to be replaced with something a bit more robust in the future
+func sqlTableIsManagedProperty(key string) bool {
+	managedProps := map[string]bool{
+		"delta.lastCommitTimestamp":                                true,
+		"delta.lastUpdateVersion":                                  true,
+		"delta.minReaderVersion":                                   true,
+		"delta.minWriterVersion":                                   true,
+		"view.catalogAndNamespace.numParts":                        true,
+		"view.catalogAndNamespace.part.0":                          true,
+		"view.catalogAndNamespace.part.1":                          true,
+		"view.query.out.col.0":                                     true,
+		"view.query.out.numCols":                                   true,
+		"view.referredTempFunctionsNames":                          true,
+		"view.referredTempViewNames":                               true,
+		"view.sqlConfig.spark.sql.hive.convertCTAS":                true,
+		"view.sqlConfig.spark.sql.legacy.createHiveTableByDefault": true,
+		"view.sqlConfig.spark.sql.parquet.compression.codec":       true,
+		"view.sqlConfig.spark.sql.session.timeZone":                true,
+		"view.sqlConfig.spark.sql.sources.commitProtocolClass":     true,
+		"view.sqlConfig.spark.sql.sources.default":                 true,
+		"view.sqlConfig.spark.sql.streaming.stopTimeout":           true,
+	}
+	return managedProps[key]
+}
+
+func (ti *SqlTableInfo) initCluster(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) (err error) {
+	defaultClusterName := "terraform-sql-table"
+	clustersAPI := clusters.NewClustersAPI(ctx, c)
```

Comment:
use the `Clusters` property from `c.WorkspaceClient()` https://github.com/databricks/databricks-sdk-go/blob/v0.7.0/workspace_client.go#L106, because `clusters.NewClustersAPI` is going away very soon in favor of SDK

Reply:
makes sense but im trying to work through some (what look like) missing pieces in the new API: 

* `GetOrCreateRunningCluster`
* `GetSmallestNodeType`
* `LatestSparkVersionOrDefault`

Reply:
This last one is waiting on a decision to use new SDK for clusters API which doesn't yet support access type (required for unity compatibility). 
Please let me know if we should wait until SDK support it or revert to legacy model and upgrade later.

Reply:
Reverted to old SDK for now to keep this one moving.

I still have the code ready locally for upgrade when we can create unity enabled clusters thought new SDK.